Add upgrade warning for `ansible_tower_fqdn` removal (#3720)
theforeman/foreman-documentation: guides/doc-Release_Notes/topics/foreman.adoc

// If this section would be empty otherwise, uncomment the following line:

> There are no upgrade warnings with Foreman {ProjectVersion}.

Reviewer: There now is an upgrade warning
```suggestion
```

For more details, see the https://community.theforeman.org/t/drop-debian-11-ruby-2-7-and-nodejs-14-support-in-foreman-3-14/40503[removal RFC].
endif::[]

> === {awx} Parameter Change

Reviewer 2: Is it a host parameter or a global setting?

Reviewer: Perhaps worth to call out AAP 2.5+ compatibility? Not sure if AWX and AAP maintain the same versioning, but something like this perhaps?
```suggestion
===  Parameter change for {awx} 2.5+ compatibility
```

Reviewer 2: I'm confused but I'm sure that Ewoud and other writers will help you to finish it.

Reviewer: > Is it a host parameter or a global setting?

It's parameters that affect hosts. That means it can be a global parameter, but also a hostgroup parameter or a host parameter. However, not a global setting.

Author: > Perhaps worth to call out AAP 2.5+ compatibility? Not sure if AWX and AAP maintain the same versioning, but something like this perhaps?

I couldn't find any references to AAP in our documentation, so I wasn't sure if we should explicitly mention it. However, specifying the supported AAP versions would make this change clearer. Should we include it?

Author: That's why I didn't explicitly mention it. If I understood you correctly, we should explicitly mention versioning for AAP to ensure clarity.

Reviewer: I think it helps readers, but wasn't quite sure how to best phrase it.

> The `ansible_tower_fqdn` parameter has been removed and replaced with `ansible_tower_api_url`. This new parameter includes both the previous `ansible_tower_fqdn` value and the required API path for {awx}.

Reviewer 3: ```suggestion
The `ansible_tower_fqdn` parameter has been removed and replaced with `ansible_tower_api_url`.
This new parameter includes both the previous `ansible_tower_fqdn` value and the required API path for {awx}.
```

> By default, the API path is set to `/api/controller/v2`. If you are using an older version of {awx} that requires a different API path, you will need to update it manually to `/api/v2`.

Reviewer: This is something I missed in the code (which is why I love user focused release notes as a secondary check): today we use `/api/v2`. This also implies that all users upgrading to Foreman 3.14 need to have AAP 2.5+ or perform a manual update.

I'm unsure how to best phrase that, but it's good to call out.

Author: Do you mean that the current phrasing is not enough?

Reviewer 3: Would one of these suggestions help?

At the very least, this is required for style review:
```suggestion
By default, the API path is set to `/api/controller/v2`.
If you are using an older version of {awx} that requires a different API path, update it manually to `/api/v2`.
```
But even better, can we be specific about the versions (is 2.3 the right one?):
```suggestion
By default, the API path is set to `/api/controller/v2`.
If you are using {awx}{nbsp}2.3, update the API path manually to `/api/v2`.
```
And according to certain resources downstream, we could also recommend users to avoid upgrading altogether:
```suggestion
By default, the API path is set to `/api/controller/v2`.
If you are using {awx}{nbsp}2.3, update the API path manually to `/api/v2` or avoid upgrading to {Project} {nbsp}3.14 until you upgrade to {awx}{nbsp}2.5.
```
The last suggestion sounds a lot like a known issue. I don't know how those are handled upstream but it could be an option too.

Reviewer 3: Is it okay that the Ansible config guide still references `ansible_tower_fqdn`? https://docs.theforeman.org/3.14/Managing_Configurations_Ansible/index-katello.html#Configuring_Provisioning_Callback_for_a_Host_ansible

Reviewer: Good catch. It isn't because that's going away. Thanks for proving the point that I often make: release notes are a great time to review a feature.

Reviewer: > But even better, can we be specific about the versions (is 2.3 the right one?):

I'd prefer a variation on this. Say something like:
```suggestion
By default, the API path is set to `/api/controller/v2` as used by {awx}{nbsp}2.5 and newer.
If you are using an older version of {awx}, update the API path manually to `/api/v2`.
```

Author: I agree, because the same applies for AAP 2.4 (using the old path `/api/v2`).

Author: After reviewing the AWX versions on https://github.com/ansible/awx/releases, I believe we need to distinguish between AWX and AAP, as they have different versioning.

Author: I couldn't find any mention of the new API path in their docs https://ansible.readthedocs.io/projects/awx/en/latest/rest_api/browseable.html.

Reviewer: I just noticed that `{awx}` expands to `Ansible Automation Controller`. That abbreviates to AAC, not AAP. Looking at https://docs.redhat.com/en/documentation/red_hat_ansible_automation_platform/2.5 I'd also expect it to be Ansible Automation Platform. It was changed in ec21fdfae90872042875e6956f73b9457df1350e.

I also get the impression that Controller is a sub-project of platform. Upstream for that part appears to be https://docs.ansible.com/automation-controller/latest/html/release-notes/index.html.

Not sure what my conclusion is, other than to be utterly confused.

Author: We can add a note advising users to verify the correct API path for their instance if they're unsure about their 
version.

Reviewer: I think that's best. Let's not spend too much time fine tuning the perfect answer here.

Reviewer 4: Also giving my 2 cents here now 🙂 
I think the differentiation has to be made more broad overall, as far as I can see in my installs of AWX, the API path never changed, this must be a AAP specific change.
At least AWX 24.6.1, which was the last version before the big rewrite, which is going on since last year, still uses `/api/v2`.

I suspect due to the multi-project nature of AAP a reverse-proxy applied sub-path got added in AAP 2.5.

For more details, see the https://community.theforeman.org/t/drop-debian-11-ruby-2-7-and-nodejs-14-support-in-foreman-3-14/40503[removal RFC].
endif::[]

=== {awx} Parameter Change
The `ansible_tower_fqdn` parameter has been removed and replaced with `ansible_tower_api_url`.
This new parameter includes both the previous `ansible_tower_fqdn` value and the required API path for {awx}.

> By default, the API path is set to `/api/controller/v2` as used by {awx}{nbsp}2.5 and newer.

Reviewer: This now expands to `AWX` in upstream (and is unused in downstream). Given the different versioning I now think it'd be best to expand it manually
```suggestion
By default, the API path is set to `/api/controller/v2` as used by Ansible Automation Controller{nbsp}2.5 and newer.
```

Author: But AAC and AAP are two different things. See https://access.redhat.com/support/policy/updates/ansible-automation-platform#packages-and-versions.

For more details, see the https://community.theforeman.org/t/drop-debian-11-ruby-2-7-and-nodejs-14-support-in-foreman-3-14/40503[removal RFC].
endif::[]

=== {awx} Parameter Change
The `ansible_tower_fqdn` parameter has been removed and replaced with `ansible_tower_api_url`.
This new parameter includes both the previous `ansible_tower_fqdn` value and the required API path for {awx}.

By default, the API path is set to `/api/controller/v2`.

> If you are using an older {awx} version, update the API path manually to `/api/v2`.

Reviewer 3: This might raise questions about which version you mean. I have no idea here either but I'm going to suggest something and hope you will correct me :) My point is to be more specific:

```suggestion
If you are using an {awx} version earlier than X.Y, update the API path manually to `/api/v2`.
```

Author: See previous discussion here https://github.com/theforeman/foreman-documentation/pull/3720#discussion_r2009961092

Reviewer 5: I dont't quite understand this. Are you saying that it does not make any sense from a tech perspective to mention the version? (aka. declining the suggestion)? Please let me know; then I will merge as is.

Author: This is the Jira ticket if you need more context https://issues.redhat.com/browse/SAT-30761
The issue is that we have no mentioning of the AAP in our docs, only AAC (Ansible Automation Controller) for downstream, or AWX for upstream.
However, this change is required for AAP only. AWX remains without any changes.

> If you're unsure which version you have, check your instance's API endpoints to confirm the correct path.

Author: No, it means the user's AAP/AWX instance.
